### ISSUE TITLE
member variable with type annotation and not used anywhere of python class  is missed

### ISSUE THREAD

**Issue by zhangzq** [doxygen/doxygen#10827]
**Describe the bug** 

The member variable of python class is missed, if:
1. with type annotation
2. not appeared elsewhere.

Below is a short example. The `a` will show in public member of docs, but `b` is missed.

```python
class Test:
    def __init__(self):
        ## first var
        self.a:int = 1

        ## second var
        self.b:int = 2

    def test(self):
        self.a = 3
```
 

**To Reproduce** 

```python
class Test:
    def __init__(self):
        ## first var
        self.a:int = 1

        ## second var
        self.b:int = 2

    def test(self):
        self.a = 3
```

with default doxyfile.

**Expected behavior** 

The class has two public members, `a` and `b`, now `b` is missed.

**Version** 
1.11.0 (74f8b41aead51679bf364056b4fd091cf828a7cf)

**Comment by albert-github** [comment]
Not only `b` is missing but also the documentation from `a` as we get the warning:
```
test.py:13: warning: Member a (variable) of class test.Test is not documented.
```

Problem here is the "typing" which originally was missing in python and now is subsequently added.

**Comment by albert-github** [comment]
Code has been integrated and corrected in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
